subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/AbstractConfigurationMetadata.java

```diff
@@ -110,7 +111,8 @@ public String getName() {
 
     @Override
     public Identifier getIdentifier() {
-        return null;
+        // TODO: This should be based on attributes?
+        return new ComponentConfigurationIdentifier(getComponentId(), name);
```

Comment:
Should this have a field for storage or be lazily computed? Is there any risk of newing many CCI objects?